packages/material-ui-lab/src/themeAugmentation/index.d.ts

```diff
@@ -0,0 +1,2 @@
+export * from './overrides';
```

Comment:
this files is not exported in the main index.d.ts

Reply:
It was decided above that that would be too magical for users and that an explicit import for `themeAugmentation` would be better.

Reply:
orite, makes sense. How exactly should we use this

e.g. this is my current theme file `theme.tsx`
```js
import createMuiTheme from "@material-ui/core/styles/createMuiTheme";
import type '@material-ui/lab/themeAugmentation'; // error: '=' expected.ts(1005)

export const theme = createMuiTheme({
  overrides: {
    // MuiAlert: {},
  }
})
```

Reply:
I am using create-react-app BTW

Reply:
Tried this
```
import type * as LabTypes from '@material-ui/lab/themeAugmentation';
```
worked but eslint is complaining that 
```
'LabTypes' is declared but its value is never read.ts(6133)
```

Reply:
Just import worked
```
import '@material-ui/lab/themeAugmentation';
```

Reply:
```tsx
import type '@material-ui/lab/themeAugmentation';
```
If you are on TS 3.8+ and your linter is up to date with TS 3.8 support. If not, then
```tsx
import '@material-ui/lab/themeAugmentation';
```

Reply:
Thank you very much. very helpful.

I googled so much but cannot find a best way for this.

What is the way to tell makeStyles that you are only allowed to use Alert classes? The following works but I am sure there must be a better way.

```typescript
import { Theme, makeStyles } from "@material-ui/core";
import { AlertClassKey } from "@material-ui/lab/Alert";

export const useAlertClasses = makeStyles<Theme>(
  (): Partial<Record<AlertClassKey, any>> => ({
    root: {
      borderRadius: 3,
    }
}));
```

Reply:
@CarsonF I'm facing the same issue. I'm also using `create-react-app`. Already made sure that all sub-dependencies are up to date by using `yarn upgrade react-scripts`. I'm using TypeScript 3.8.3.
Named type imports are working completely fine but not these unnamed ones.

Reply:
@CarsonF I'm facing the same issue. I'm also using `create-react-app`. Already made sure that all sub-dependencies are up to date by using `yarn upgrade react-scripts`. I'm using TypeScript 3.8.3.
Named type imports are working completely fine but not these unnamed ones.

Reply:
It should be fixed in v5.0.0-alpha.3 with #21724

Reply:
@oliviertassinari Could you backport this fix to the 4.x line of labs? I don't want to upgrade to 5.x yet.

Reply:
@levrik For now, you can copy & paste the diff of this pull request